lab3/DeleteFoodAdminServlet.java

```diff
@@ -0,0 +1,27 @@
+package lab3;
+
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/admin/foods/delete")
+public class DeleteFoodAdminServlet extends HttpServlet {
+    public void doGet( HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        int id = Integer.parseInt(request.getParameter("id"));
+        List<FoodEntry> entries = (List<FoodEntry>) getServletContext().getAttribute("entries");
+        int index = -1;
+        for (int i = 0; i < entries.size(); i ++) {
+            if (entries.get(i).getId() == id) {
+                index = i;
+            }
+        }
+        entries.remove(index);
+        getServletContext().setAttribute("entries", entries);
```

Comment:
As reminder, all Servlets are sharing the same application scope. This implies you should name the key-name more explicit (e.g. "food-items")  in order to separate out different applications (e.g. midterm and lab3)